Infrastructure/Services/EmailService.cs

```diff
@@ -10,25 +10,28 @@ namespace Infrastructure.Services;
 public class EmailService : IEmailService
 {
     private readonly ISendGridClient _client;
-    private readonly string _sendGridApiKey;
     private readonly string _sendGridSenderEmail;
```

Comment:
Do not pass all the properties separately try to use IOption pattern 
https://andrewlock.net/how-to-use-the-ioptions-pattern-for-configuration-in-asp-net-core-rc2/